syntax = "proto3";


message CoffeeCommand {
    enum CommandType {
        OPERATION       = 0;
        QUERY           = 1;
    }

    CommandType type    = 1;
    bytes command       = 2;
}

message Response {
    enum ResponseType {
        OK              = 0;
        RESULT          = 1;
        OPERATION_ERR   = 2;
        FORMAT_ERR      = 3;
    }

    message ResultTable {
        bool POWER      = 1;
        bool WATER      = 2;
        bool BEANS      = 3;
        bool TRAY       = 4;
    }

    ResponseType type   = 1;
    string description  = 2;
    ResultTable results = 3;
}
